docs/guides/reverification.mdx

```diff
@@ -0,0 +1,205 @@
+---
+title: Add reverification for sensitive actions
+description: Learn how to implement Clerk's reverification feature to protect sensitive actions in your application.
+---
+
+> [!WARNING]
+> This feature is currently in public beta. **It is not recommended for production use**.
+>
+> Depending on the SDK you're using, this feature requires `@clerk/nextjs@6.5.0` or later, `@clerk/clerk-sdk-ruby@3.3.0` or later, and `@clerk/clerk-js@5.35.0` or later.
+
+Reverification allows you to prompt a user to verify their credentials before performing sensitive actions, even if they're already authenticated. For example, in a banking application, transferring money is considered a "sensitive action." Reverification can be used to confirm the user's identity.
+
+## How to require reverification
+
+To implement reverification, you need to handle it both on the server- and client-side.
+
+### Handle reverification server-side
+
+To handle reverification server-side, use the [`auth.has()`](/docs/references/nextjs/auth-object#has) helper to check if the user has verified their credentials within a specific time period. Pass a [configuration](#supported-verification-configurations) to set the time period you would like. If the user hasn't verified their credentials within that time period, return either `reverificationError` or `reverificationErrorResponse`, depending on the framework you're using. Use the following tabs to see examples for different frameworks.
+
+> [!WARNING]
+> `reverificationErrorResponse` and `reverificationError` requires `@clerk/shared@2.17.0` or later, and `@clerk/clerk-js@5.35.0` or later.
+
+<Tabs items={["Next.js Server Action", "Next.js Route Handler", "Ruby", "Other"]}>
+  <Tab>
+    The following example uses the [`has()`](/docs/references/nextjs/auth-object#has) helper to check if the user has verified their credentials within a specific time period. The `strict` configuration sets the time period to 10 minutes. If the user hasn't verified their credentials within 10 minutes, the `reverificationError` utility is used to return an error.
+
+    > [!NOTE]
+    > Next.js Server Actions use `reverificationError` instead of `reverificationErrorResponse` as they need to return a serializable object. A serializable object refers to data that can be safely converted to a JSON format for transmission between the server and client.
```

Comment:
I believe this is not necessary, since the snippet indicates that already. They will also get an error that the can't return a response from a server action